src/a-runtime-console/kubernetes/store/oauth-config-store.spec.ts

```diff
@@ -0,0 +1,283 @@
+    },
+    id: 'mock',
+    type: 'mock'
+  };
+
+  let data = {};
+
+  let subscription: Subscription;
+
+  beforeEach(() => {
+    mockLogger = jasmine.createSpyObj<Logger>('Logger', ['error']);
+    mockErrorHandler = jasmine.createSpyObj<ErrorHandler>('ErrorHandler', ['handleError']);
+    mockNotificationsService = jasmine.createSpyObj<NotificationsService>('NotificationsService', ['message']);
+  });
+
+  describe('success state', () => {
+    beforeEach(() => {
+      mockUserService = createMock(UserService);
+      mockUserService.loggedInUser = Observable.of(user).publish();
+
+      TestBed.configureTestingModule({
+        imports: [HttpModule],
+        providers: [
+          {
+            provide: XHRBackend, useClass: MockBackend
+          },
+          {
+            provide: UserService, useClass: mockUserService
+          },
+          {
+            provide: Logger, useValue: mockLogger
+          },
+          {
+            provide: ErrorHandler, useValue: mockErrorHandler
+          },
+          {
+            provide: NotificationsService, useValue: mockNotificationsService
+          },
+          {
+            // provide OAuthConfigStore with a factory inside the fakeAsync zone
+            // so the mockBackend can catch http requests made inside the constructor
+            provide: OAuthConfigStore, useFactory: fakeAsync((
+              http: Http,
+              mockBackend: MockBackend,
+              logger: Logger,
+              errorHandler: ErrorHandler,
+              notifications: NotificationsService
+            ) => {
+              mockBackend.connections.subscribe((connection: MockConnection) => {
+                connection.mockRespond(new Response(new ResponseOptions({
+                  body: JSON.stringify(data),
+                  status: 200
+                })));
+              });
+              return new OAuthConfigStore(http, mockUserService, logger, errorHandler, notifications);
+            }),
+            deps: [Http, XHRBackend, Logger, ErrorHandler, NotificationsService]
+          }
+        ]
+      });
+
+      mockBackend = TestBed.get(XHRBackend);
+      oauthStore = TestBed.get(OAuthConfigStore);
+    });
+
+    it('should load and set latest oauthconfig on init', (done: DoneFn) => {
+      mockUserService.loggedInUser.connect();
+
+      oauthStore.loading.subscribe((val: boolean) => {
+        if (!val) {
+          oauthStore.resource.subscribe((config: OAuthConfig) => {
+            expect(config.loaded).toBeTruthy();
+            done();
+          });
+        }
+      });
+    });
+
+    it('should eventually set openshift console on init', (done: DoneFn) => {
+      mockUserService.loggedInUser.connect();
+
+      oauthStore.loading.subscribe((val: boolean) => {
+        if (!val) {
+          oauthStore.resource.subscribe((config: OAuthConfig) => {
+            expect(config.loaded).toBeTruthy();
+            if (config.openshiftConsoleUrl) {
```

Comment:
Same idea here with the `config.openShiftConsoleUrl` check - this should probably become an `expect().toBeTruthy()`.